apps/tailwind-components/components/input/Listbox.vue

```diff
@@ -0,0 +1,388 @@
+  btnElemRef.value?.focus();
+}
+
+function openCloseListbox() {
+  isExpanded.value = !isExpanded.value;
+  if (isExpanded.value) {
+    ulElemRef.value?.setAttribute("tabindex", "0");
+    if (modelValue.value) {
+      updateCounter(startingCounter.value);
+    } else {
+      updateCounter(0);
+    }
+    focusListOption();
+  } else {
+    ulElemRef.value?.setAttribute("tabindex", "-1");
+  }
+}
+
+function onListboxButtonKeyDown(event: KeyboardEvent) {
+  const key = event.key;
+  switch (key) {
+    case "ArrowUp":
+      openCloseListbox();
+      focusPreviousOption();
+      break;
+
+    case "ArrowDown":
+      openCloseListbox();
+      break;
+
+    case "Enter":
+      openCloseListbox();
+      break;
+
+    case "Space":
+      openCloseListbox();
+      break;
+
+    case "Home":
+      focusCounter.value = 0;
+      focusListOption();
+      break;
+
+    case "End":
+      focusCounter.value = listboxOptions.value.length - 1;
+      focusListOption();
+      break;
+  }
+}
+
+function onListboxKeyDown(event: KeyboardEvent) {
+  const key = event.key;
+  switch (key) {
+    case "ArrowUp":
+      focusPreviousOption();
+      break;
+
+    case "ArrowDown":
+      focusNextOption();
+      break;
+
+    case "PageUp":
+      focusPreviousOption(10);
+      break;
+
+    case "PageDown":
+      focusNextOption(10);
+      break;
+
+    case "Home":
+      focusCounter.value = 0;
+      focusListOption();
+      break;
+
+    case "End":
+      focusCounter.value = listboxOptions.value.length - 1;
+      focusListOption();
+      break;
+  }
+}
+
+function onListboxOptionClick(option: IInternalListboxOption) {
+  updateModelValue(option);
+  focusListboxButton();
+}
+
+function onListboxOptionKeyDown(
+  event: KeyboardEvent,
+  option: IInternalListboxOption
+) {
+  const key: string = event.key;
+  switch (key) {
+    case "Enter":
+      updateModelValue(option);
+      focusListboxButton();
+      break;
+
+    case "Spacebar":
+      updateModelValue(option);
+      focusListboxButton();
+      break;
+
+    // spacebar (for older browser support)
+    case " ":
+      updateModelValue(option);
+      focusListboxButton();
+      break;
+
+    case "Tab":
+      updateModelValue(option);
+      focusListboxButton();
+      break;
+
+    case "Escape":
+      blurListOption(option);
+      openCloseListbox();
+      focusListboxButton();
+      break;
+  }
+}
+
+onMounted(() => {
+  if (modelValue.value && listboxOptions.value) {
+    const initalValue: IInternalListboxOption[] = listboxOptions.value.filter(
+      (row: IInternalListboxOption) => {
+        return (
+          row.value === (modelValue.value as IListboxOption).value ||
+          row.value === (modelValue.value as IListboxValue)
+        );
+      }
+    );
+    if (initalValue) {
+      updateModelValue(initalValue[0], false);
+    }
+  }
+});
+
+watch(
+  () => props.placeholder,
+  () => {
+    displayText.value = props.placeholder;
+  }
+);
```

Comment:
I suggest putting the watch and onMounted more to the top of the file, now they feel like an afterthought, whilst they are quite important when looking at this file. For readability you could maybe extract the anonymous functions and put those here at the end.